File: src/main/java/pl/training/workshop/m5/s05_extractsubclass/step4/Programme.java
package pl.training.workshop.m5.s05_extractsubclass.step4;

/** Krok 4: bez zmian - klient przeżył całą ekstrakcję bez modyfikacji od kroku 1. */
public final class Programme {
    public String line(String title, String format, String guest) {
        Screening screening = guest == null
                ? Screening.regular(title, format)
                : Screening.premiere(title, format, guest);
        return screening.describe() + " | " + screening.price();
    }
}

File: typescript/src/workshop/m5/s05_extractsubclass/step2/Programme.ts
import { Screening } from './Screening.js';

/** Krok 2: bez zmian - klient nie wie, że fabryka zwraca podklasę. */
export class Programme {
  line(title: string, format: string, guest: string | null): string {
    const screening: Screening = guest === null
      ? Screening.regular(title, format)
      : Screening.premiere(title, format, guest);
    return screening.describe() + ' | ' + screening.price();
  }
}
